Wyam.Core/Modules/CopyFiles.cs

```diff
@@ -106,7 +106,7 @@ public IEnumerable<IDocument> Execute(IReadOnlyList<IDocument> inputs, IExecutio
                             .Select(file =>
                             {
                                 string destination = _destinationPath == null
-                                    ? Path.Combine(context.OutputFolder, PathHelper.GetRelativePath(Path.GetDirectoryName(path), Path.GetDirectoryName(file)), Path.GetFileName(file)) 
+                                    ? Path.Combine(context.OutputFolder, PathHelper.GetRelativePath(context.InputFolder, Path.GetDirectoryName(file)), Path.GetFileName(file)) 
```

Comment:
Ok, I will update the PR accordingly.

> In fact, one thing I want to look at soon is a better abstraction for the file system to take the place of System.IO usages. I especially like the one used in Cake, so may take some inspiration there.

That would be nice. That should also make unit testing easier.
As I saw, Cake uses static classes for the abstraction. I would prefer a more interface-based approach. Have you thought about bringing dependency injection into Wyam?